### PR TITLE
[WIP] FWI-1646 - Trivy scan is not "rolling"

### PR DESCRIPTION
**Pull request by jdesouza**
This PR fixes #

## Checklist
* [ ] I have signed the CLA
* [ ] I have updated/added any relevant documentation

### What alternative solution should we consider, if any?

[Internal Ticket FWI-1646](https://fairwinds.myjetbrains.com/youtrack/issue/FWI-1646)

### CODE REVIEW

File: plugins/ci/CHANGELOG.md [hunk 1]
```diff
@@ -1,7 +1,8 @@
 # Changelog
 
-## 1.4.2
-* Update alpine to remove CVE
+
+## 1.5.0
+* Trivy no longer downloads images
```
Comment (rbren) [inline plugins/ci/CHANGELOG.md:5]: looks like we lost the 1.4.2 entry

File: plugins/trivy/cmd/trivy/main.go [hunk 2]
```diff
@@ -30,95 +30,20 @@ type newestVersions struct {
...
-				break
-			}
-		}
-		if keep {
-			imagesToKeep = append(imagesToKeep, report)
-		}
-	}
-	lastReport.Images = imagesToKeep
+	imagesToScan := getImagesToScan(images, lastReport.Images)
+	lastReport.Images = getImagesToKeep(images, lastReport, imagesToScan)
```
Comment (rbren) [inline plugins/trivy/cmd/trivy/main.go:41]: can we log both `images` and `imagesToScan`?

File: plugins/trivy/cmd/trivy/main.go [hunk 3]
```diff
@@ -174,3 +105,93 @@ func getNewestVersions(versionsChan chan newestVersions, ctx context.Context, im
...
+		for _, report := range lastReportImages {
+			reportSha := getShaFromID(report.ID)
+			imageSha := getShaFromID(image.ID)
+			if report.Name == image.Name && reportSha == imageSha {
+				found = true
+				break
+			}
+		}
+		if !found {
+			imagesToScan = append(imagesToScan, image)
```
Comment (rbren) [inline plugins/trivy/cmd/trivy/main.go:130]: I think we need `if len(imagesToScan) < numberToScan {`

File: plugins/ci/CHANGELOG.md (hunk 1 above)
Comment (jdesouza) [reply plugins/ci/CHANGELOG.md:5]: Fixed

File: plugins/trivy/cmd/trivy/main.go (hunk 3 above)
Comment (jdesouza) [reply plugins/trivy/cmd/trivy/main.go:130]: I was doing that after the call, but I believe I had a bug. Glad about this comment

File: plugins/trivy/cmd/trivy/main.go (hunk 2 above)
Comment (jdesouza) [reply plugins/trivy/cmd/trivy/main.go:41]: done

**Review by rbren — APPROVED** [review]

**Pull request merged**